### PR TITLE
Sample App walkthrough

### PR DESCRIPTION
**Pull request by tom-rosewell**
WIP

### CODE REVIEW

**Comment by tom-rosewell** [comment]
I still need to fix the search endpoint and document it

**Comment by tom-rosewell** [comment]
Search endpoint has been fixed

**Comment by tom-rosewell** [comment]
Should now be finished.

File: modules/hello-world/examples/sample-app.py [hunk 1]
```diff
@@ -0,0 +1,478 @@
+# REFERENCE FILE ONLY.
+# This file is a copy of 'travel.py' in the 'try-cb-python' repo. It's not
```
Comment (RichardSmedley) [inline modules/hello-world/examples/sample-app.py:2]: This file will need to become a 7.2 version of https://github.com/couchbaselabs/try-cb-python/blob/7.1/travel.py - we'll need a mechanism to keep them in sync - but I'm not going to block on that, so we'll call it a future todo.

File: modules/hello-world/pages/sample-application.adoc [hunk 2]
```diff
@@ -15,47 +16,287 @@ include::{version-common}@sdk:shared:partial$sample-application.adoc[tag=abstrac
...
+
+----
+{
+  'username':<username>,
+  'password':<password>
+}
+----
+
+NOTE: The document key is a lowercase version of the provided username - this means that different usernames can refer to the same document. 
+For example, if a user signs up with the username `Douglas Reynholm`, a different user can't sign up with the username `douglas reynholm`, as both resolve to the same document key.
```
Comment (RichardSmedley) [inline modules/hello-world/pages/sample-application.adoc:121]: No getting rid of Reynholm industries  :)

**Review by RichardSmedley — APPROVED** [review]
Great stuff @tom-rosewell - much more useful now.  :+1:

**Pull request merged**